jetty-server/src/main/java/org/eclipse/jetty/server/HttpOutput.java

```diff
@@ -378,6 +378,19 @@ private boolean updateApiState(Throwable failure)
         return wake;
     }
 
+    private int getAggregateSpace()
+    {
+        // If no aggregate, we can allocate one of bufferSize
+        if (_aggregate == null)
+            return getBufferSize();
+
+        // if the position is not zero, compact to avoid empty at capacity
+        if (_aggregate.position() != 0)
+            BufferUtil.compact(_aggregate);
+
+        return BufferUtil.space(_aggregate);
```

Comment:
How can the calculation of space not be based on capacity?

The spec allows for a larger buffer than set:
```
     * The servlet container will use a buffer at least as large as 
     * the size requested.  The actual buffer size used can be found
     * using <code>getBufferSize</code>.
```
It would be crazy to allocate a larger buffer and then run a shadow capacity to only use a part of it.  If you think we should not use larger buffers than requested, then the buffer pool should never return a larger buffer than requested.

Reply:
I agree with @sbordet here.

If a user uses `HttpConfiguration.setOutputBufferSize(512)` or `HttpConfiguration.setOutputAggregationSize(256)` then they expect behavior based on that configuration, not some larger / mysterious / black box / implementation specific size.

The fact that our ByteBufferPool _might_ return a larger size is a side-effect of being efficient with memory allocation here.
I would expect our behavior to remain unchanged, using the returned ByteBuffer (regardless of actual space/capacity) for the configured buffer size (eg: 512).

Reply:
Sorry but it is nonsensical to return a larger buffer and then only use a part of it.  The contract allows for a larger buffer to be used, so why would we not use it if it has been allocated?